x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpExporter.java

```diff
@@ -401,6 +405,7 @@ public void validate(String password, Map<Setting<?>, Object> settings) {
      */
     private final AtomicBoolean clusterAlertsAllowed = new AtomicBoolean(false);
 
+    private static final Map<String, String> SECURE_AUTH_PASSWORDS = new HashMap<>();
```

Comment:
I could change it to a `Map<String, SecureString>` but I couldn't close any of the `SecureString` instances because the HTTP exporter has to be recreated (so I need the full value, not just an equality check) any time any of its settings, whether secure or not, change. Unfortunately, this is unavoidable for settings that are both secure and reloadable since the keystore itself is not accessible outside of node startup and reload events. The security folks directed me to 

https://github.com/elastic/elasticsearch/blob/c3b5f713e2a9fdfc6a9f05892aa7647d905b55d4/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/NotificationService.java#L167

where they did essentially the same thing for the secure settings in watcher notifications.